src/attr/_make.py

```diff
@@ -70,6 +70,30 @@ def __repr__(self):
 """
 
 
+class _CacheHashWrapper(int):
+    """An integer subclass that pickles / copies as None
+
+    This is used for non-slots classes with ``cache_hash=True``, to avoid
+    serializing a potentially (even likely) invalid hash value. Since ``None``
+    is the default value for uncalculated hashes, whenever this is copied,
+    the copy's value for the hash should automatically reset.
+
+    See GH #613 for more details.
+    """
+
+    if PY2:
+        # For some reason `type(None)` isn't callable in Python 2, but we don't
+        # actually need a constructor for None objects, we just need any
+        # available function that returns None.
```

Comment:
FWIW, Python 2.7 has `types.NoneType`:

```console
Python 2.7.16 (default, Nov  9 2019, 05:55:08) 
[GCC 4.2.1 Compatible Apple LLVM 11.0.0 (clang-1100.0.32.4) (-macos10.15-objc-s on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> type(None)
<type 'NoneType'>
>>> import types
>>> types.NoneType
<type 'NoneType'>
```

Interestingly, while `NoneType` in Python3 is a thing, I don't know how to reach it:

```console
Python 3.8.0 (v3.8.0:fa919fdf25, Oct 14 2019, 10:23:27) 
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> type(None)
<class 'NoneType'>
>>> import types
>>> types.NoneType
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'types' has no attribute 'NoneType'
>>> 
```

Reply:
`types.NoneType` isn't helpful to us here in Python 2, since it isn't callable:

```
>>> types.NoneType()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: cannot create 'NoneType' instances
```

Why that doesn't just return the singleton instance is beyond me, but oh well...

Reply:
Ah, yes.